calva/repl/middleware/testRunner.ts

```diff
@@ -7,85 +7,135 @@ import * as util from '../../utilities';
-    _.each(responsesArray, (responses) => {
-        _.each(responses, response => {
-            let results = response.results || null,
-                summary = response.summary || null;
-            if (results !== null) {
-                _.each(results, (tests, ns) => {
-                    _.each(tests, (asserts, test) => {
-                        _.each(asserts, a => {
-                            if (a.type == "error") {
-                                if (log) {
-                                    chan.appendLine("ERROR in: " + ns + ": " + a.file + ", line " + a.line +
-                                        ": " + test + ": " + (a.context || "") + ":\n" +
-                                        "  error: " + a.error + "\n  expected: " + a.expected);
-                                }
-                            }
-                            if (a.type == "fail") {
-                                let msg = "failure in test: " + test +
-                                    " context: " + a.context + ", expected " +
-                                    a.expected + ", got: " + a.actual,
-                                    err = new vscode.Diagnostic(new vscode.Range(a.line - 1, 0, a.line - 1, 1000),
-                                        msg,
-                                        vscode.DiagnosticSeverity.Error);
-                                if (!diagnostics[a.file]) {
-                                    diagnostics[a.file] = [];
-                                }
-                                diagnostics[a.file].push(err);
-                                if (log) {
-                                    chan.appendLine("FAIL in: " + a.file + ":" + a.line +
-                                        ": " + test + ": " + (a.context || "") + ":\n" +
-                                        "  expected: " + a.expected + "\n  actual: " + a.actual);
-                                }
-                            }
-                        })
-                    })
-                })
-            }
-            if (summary !== null) {
-                _.each(summary, (v, k) => {
-                    total_summary[k] = summary[k] + (total_summary[k] !== undefined ? total_summary[k] : 0);
-                });
-            }
-        });
-    });
-    if (total_summary !== null) {
-        let hasProblems = total_summary.error + total_summary.fail > 0;
-        if (log) {
-            chan.appendLine("\n" + (total_summary.test > 0 ?
-                total_summary.test + " tests finished, " +
-                (!hasProblems ? "all passing 👍" :
-                    "problems found. 😭" +
-                    " errors: " + total_summary.error + ", failures: " + total_summary.fail) :
-                "No tests found. 😱") +
-                ", ns: " + total_summary.ns + ", vars: " + total_summary.var);
+interface Response {
+    session: String
+    id? : String
+    status? : String[]
+}
+
+interface TestResponse extends Response {
```

Comment:
I defined some types to try to make it easier to deal with nREPL messages:

https://clojurians.slack.com/archives/C17JYSA3H/p1542971155229800